pkg/distro/generic/options_test.go

```diff
@@ -0,0 +1,1671 @@
+					},
+				},
+			},
+			// NOTE: this is listed as supported in the checks that are
+			// specific to the image type but the image type is not listed as
+			// supporting installer customizations later in the function
+			expErr: "installer customizations are not supported for \"workstation-live-installer\"",
+		},
+		"live-installer-unsupported-customizations": {
+			it: "workstation-live-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					User: []blueprint.UserCustomization{{Name: "root"}},
+					Installer: &blueprint.InstallerCustomization{
+						Unattended: true,
+					},
+				},
+			},
+			expErr: "image type \"workstation-live-installer\" does not support customizations",
+		},
+
+		"ostree-types-no-oscap": {
+			it: "iot-container",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					OpenSCAP: &blueprint.OpenSCAPCustomization{
+						ProfileID: "xccdf_org.ssgproject.content_profile_ospp",
+					},
+				},
+			},
+			expErr: "OpenSCAP customizations are not supported for ostree types",
+		},
+
+		"iot-installer-installer-customizations": {
+			it: "iot-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					Installer: &blueprint.InstallerCustomization{
+						Unattended: true,
+					},
+				},
+			},
+			options: distro.ImageOptions{
+				OSTree: &ostree.ImageOptions{
+					URL: "https://example.org/repo",
+				},
+			},
+		},
+		"iot-installer-bad-combinations": {
+			it: "iot-installer",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					User: []blueprint.UserCustomization{{Name: "root"}},
+					Installer: &blueprint.InstallerCustomization{
+						Kickstart: &blueprint.Kickstart{
+							Contents: "echo 'Testing'",
+						},
+					},
+				},
+			},
+			options: distro.ImageOptions{
+				OSTree: &ostree.ImageOptions{
+					URL: "https://example.org/repo",
+				},
+			},
+			expErr: "iot-installer installer.kickstart.contents are not supported in combination with users or groups",
+		},
+
+		"ostree-disk-unsupported-containers": {
+			it: "iot-qcow2",
+			bp: blueprint.Blueprint{
+				Containers: []blueprint.Container{
+					{
+						Source: "example.org/containers/test:42",
+					},
+				},
+			},
+			expErr: "embedding containers is not supported for iot-qcow2 on fedora-42",
+		},
+
+		"ostree-commit-unsupported-kernel-append": {
+			it: "iot-commit",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					Kernel: &blueprint.KernelCustomization{
+						Append: "debug",
+					},
+				},
+			},
+			expErr: "kernel boot parameter customizations are not supported for ostree types",
+		},
+
+		"oscap-empty-profile": {
+			it: "vhd",
+			bp: blueprint.Blueprint{
+				Customizations: &blueprint.Customizations{
+					OpenSCAP: &blueprint.OpenSCAPCustomization{
+						ProfileID: "",
+					},
+				},
+			},
+			expErr: "OpenSCAP profile cannot be empty",
+		},
```

Comment:
This is something that this validation should not care about and will be done by UBP schema. Can't hurt tho, also UBP is not yet implemented.